src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,2233 @@
+      step_mem->fe(ark_mem->tcur + ((sunrealtype)j - ONE) * sm1inv * ark_mem->h,
+                   ark_mem->ycur, ark_mem->tempv2, ark_mem->user_data);
+    step_mem->nfe++;
+    if (retval < 0) { return ARK_RHSFUNC_FAIL; }
+    if (retval > 0) { return RHSFUNC_RECVR; }
+
+#ifdef SUNDIALS_LOGGING_EXTRA_DEBUG
+    SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                       "ARKODE::lsrkStep_TakeStepSSPs2", "stage RHS",
+                       "F_%i(:) =", j);
+    N_VPrintFile(ark_mem->tempv2, ARK_LOGGER->debug_fp);
+#endif
+
+    N_VLinearSum(ONE, ark_mem->ycur, sm1inv * ark_mem->h, ark_mem->tempv2,
+                 ark_mem->ycur);
+    if (!ark_mem->fixedstep)
+    {
+      N_VLinearSum(ONE, ark_mem->tempv1, bt2 * ark_mem->h, ark_mem->tempv2,
+                   ark_mem->tempv1);
+    }
+
+    /* apply user-supplied stage postprocessing function (if supplied) */
+    if (ark_mem->ProcessStage != NULL)
+    {
+      retval = ark_mem->ProcessStage(ark_mem->tcur + ((sunrealtype)j - ONE) *
+                                                       sm1inv * ark_mem->h,
+                                     ark_mem->ycur, ark_mem->user_data);
+      if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
+    }
+  }
+
+#if SUNDIALS_LOGGING_LEVEL >= SUNDIALS_LOGGING_DEBUG
+  SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                     "ARKODE::lsrkStep_TakeStepSSPs2", "start-stage",
+                     "step = %li, stage = %i, h = %" RSYM ", tcur = %" RSYM,
+                     ark_mem->nst, step_mem->req_stages, ark_mem->h,
+                     ark_mem->tcur + ark_mem->h);
+#endif
+
+  /* Evaluate the last stage for j = step_mem->req_stages */
+  retval = step_mem->fe(ark_mem->tcur + ark_mem->h, ark_mem->ycur,
+                        ark_mem->tempv2, ark_mem->user_data);
+  step_mem->nfe++;
+  if (retval < 0) { return ARK_RHSFUNC_FAIL; }
+  if (retval > 0) { return RHSFUNC_RECVR; }
+
+#ifdef SUNDIALS_LOGGING_EXTRA_DEBUG
+  SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                     "ARKODE::lsrkStep_TakeStepSSPs2", "stage RHS",
+                     "F_%i(:) =", step_mem->req_stages);
+  N_VPrintFile(ark_mem->tempv2, ARK_LOGGER->debug_fp);
+#endif
+
+  cvals[0] = ONE / (sm1inv * rs);
+  Xvecs[0] = ark_mem->ycur;
+  cvals[1] = ONE / rs;
+  Xvecs[1] = ark_mem->yn;
+  cvals[2] = ark_mem->h / rs;
+  Xvecs[2] = ark_mem->tempv2;
+
+  retval = N_VLinearCombination(step_mem->nfusedopvecs, cvals, Xvecs,
+                                ark_mem->ycur);
+  if (!ark_mem->fixedstep)
+  {
+    N_VLinearSum(ONE, ark_mem->tempv1, bt3 * ark_mem->h, ark_mem->tempv2,
+                 ark_mem->tempv1);
+  }
+
+  /* apply user-supplied stage postprocessing function (if supplied) */
+  if (ark_mem->ProcessStage != NULL)
+  {
+    retval = ark_mem->ProcessStage(ark_mem->tcur + ark_mem->h, ark_mem->ycur,
+                                   ark_mem->user_data);
+    if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
+  }
```

Comment:
This is the new solution so the final postprocessing will happen in `arkCompleteStep`
```suggestion
```